src/main/java/io/xpring/ilp/model/AccountBalanceResponse.java

```diff
@@ -0,0 +1,86 @@
+
+import java.math.BigInteger;
+
+/**
+ * Response object for requests to get an account's balance
+ */
+@Value.Immutable
+public interface AccountBalanceResponse {
+
+  static ImmutableAccountBalanceResponse.Builder builder() {
+    return ImmutableAccountBalanceResponse.builder();
+  }
+
+  /**
+   * @return The accountId for this account balance.
+   */
+  String accountId();
+
+  /**
+   * Currency code or other asset identifier that this account's balances will be denominated in
+   */
+  String assetCode();
+
+  /**
+   * Interledger amounts are integers, but most currencies are typically represented as # fractional units, e.g. cents.
+   * This property defines how many Interledger units make # up one regular unit. For dollars, this would usually be set
+   * to 9, so that Interledger # amounts are expressed in nano-dollars.
+   *
+   * @return an int representing this account's asset scale.
+   */
+  int assetScale();
+
+  /**
+   * The amount of units representing the aggregate position this Connector operator holds with the account owner. A
+   * positive balance indicates the Connector operator has an outstanding liability (i.e., owes money) to the account
+   * holder. A negative balance represents an asset (i.e., the account holder owes money to the operator). This value is
+   * the sum of the clearing balance and the prepaid amount.
+   *
+   * @return An {@link BigInteger} representing the net clearingBalance of this account.
+   */
+  @Value.Derived
+  default BigInteger netBalance() {
+    return clearingBalance().add(prepaidAmount());
+  }
+
+  /**
+   * The amount of units representing the clearing position this Connector operator holds with the account owner. A
+   * positive clearing balance indicates the Connector operator has an outstanding liability (i.e., owes money) to the
+   * account holder. A negative clearing balance represents an asset (i.e., the account holder owes money to the
+   * operator).
+   *
+   * @return An {@link BigInteger} representing the net clearing balance of this account.
+   */
+  BigInteger clearingBalance();
+
+  /**
+   * The number of units that the account holder has prepaid. This value is factored into the value returned by {@link
+   * #netBalance()}, and is generally never negative.
+   *
+   * @return An {@link BigInteger} representing the number of units the counterparty (i.e., owner of this account) has
+   * prepaid with this Connector.
+   */
+  BigInteger prepaidAmount();
+
+  /**
+   * Constructs an {@link AccountBalanceResponse} from a {@link GetBalanceResponse}
+   *
+   * @param getBalanceResponse a {@link GetBalanceResponse} (protobuf object) whose field values will be used
+   *                           to construct an {@link AccountBalanceResponse}
+   * @return an {@link AccountBalanceResponse} with its fields set via the analogous protobuf fields.
+   */
+  static AccountBalanceResponse from(GetBalanceResponse getBalanceResponse) {
+    return builder()
```

Comment:
Note that primitives in protobuf are going to return default values. I.E. if `assetCode` is not set, it'll be set to "". 

As long as you are *sure* all of these fields are always set, that's fine. If you are not sure, then you could make this method safer by:

```
if (!getBalanceResponse.hasAssetCode()) {
  throw malformedResponse
}
assetCode = getBalanceResponse.getAssetCode()
```

I'm leaving it up to you to decide the level of safety you want to have

Reply:
I'm completely sure that they will always be set. In Hermes, we populate an immutable object (in this case, an `AccountBalanceResponse`) from a network call to the connector, then convert it to a gRCP `GetBalanceResponse`.  All of the fields in this class are required, so none can be set to `null` and therefor `GetBalanceResponse` will not have any default values.